module/VuFind/src/VuFind/ApiKey/ApiKeyTrait.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * Api Key trait.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  ApiKey
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+
+namespace VuFind\ApiKey;
+
+use Laminas\Http\Response;
+
+use function in_array;
+use function is_callable;
+
+/**
+ * Api Key trait.
+ *
+ * @category VuFind
+ * @package  ApiKey
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */
+trait ApiKeyTrait
+{
+    /**
+     * Is api key used?
+     * - 'disabled' => Keys are not used and cannot be created
+     * 'enabled' => Keys can be created and used but not enforced
+     * 'enforced' => Keys are enforced
+     *
+     * @var string
+     */
+    protected string $apiKeyMode;
+
+    /**
+     * Request header field name for the API key
+     *
+     * @var string
+     */
+    protected string $apiKeyHeaderField = '';
+
+    /**
+     * Api key service
+     *
+     * @var ApiKeyService
+     */
+    protected ApiKeyService $apiKeyService;
+
+    /**
+     * Key for header to look for an api key
+     *
+     * @var string
+     */
+    protected string $apiKeyHeader;
+
+    /**
+     * Log requests with API keys
+     *
+     * @var bool
+     */
+    protected bool $logApiKeyRequests = false;
+
+    /**
+     * Set Api Key service
+     *
+     * @param ApiKeyService $apiKeyService Api Key service
+     *
+     * @return void
+     */
+    protected function setApiKeyService(ApiKeyService $apiKeyService): void
+    {
+        $this->apiKeyService = $apiKeyService;
+    }
+
+    /**
+     * Set api key mode.
+     *
+     * @param $apiKeyMode Api key mode, disabled, enabled, enforced.
```

Comment:
Maybe this punctuation would be more clear.

```suggestion
     * @param $apiKeyMode Api key mode: disabled, enabled, enforced.
```